### PR TITLE
SMPP connector added.

### PR DESCRIPTION
**Pull request by Biruntha**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,32 @@
+# Ignore everything in this directory
```
Comment (malakasilva) [inline .gitignore]: Avoid committing this file to repo.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+    <version>1.0.0</version>
+    <packaging>bundle</packaging>
+    <name>WSO2 Carbon - Connector For SMPP</name>
+    <url>http://wso2.org</url>
+    <properties>
+        <axis2.osgi.version.range>1.7.1</axis2.osgi.version.range>
+        <axiom.osgi.version.range>1.2.5</axiom.osgi.version.range>
+        <connector.name>SMPP</connector.name>
+        <test.framework.version>4.2.0</test.framework.version>
+        <esb.version>5.0.0-BETA</esb.version>
```
Comment (malakasilva) [inline pom.xml]: Remove beta and snapshot versions and only have released versions.

File: src/main/java/org/wso2/carbon/esb/connector/SMPPConstants.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.esb.connector;
+
+public class SMPPConstants {
+    /**
```
Comment (malakasilva) [inline src/main/java/org/wso2/carbon/esb/connector/SMPPConstants.java]: Just have line comments here

File: src/main/java/org/wso2/carbon/esb/connector/SMSDto.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
+/*
```
Comment (malakasilva) [inline src/main/java/org/wso2/carbon/esb/connector/SMSDto.java]: Rename Dto to DTO.

File: src/main/resources/smppSendSMS/component.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+
```
Comment (malakasilva) [inline src/main/resources/smppSendSMS/component.xml]: Extra lines here

File: src/test/resources/artifacts/AXIS2/config/test_axis2_server_9000.xml [hunk 6]
```diff
@@ -0,0 +1,378 @@
+<!--
+  ~  Licensed to the Apache Software Foundation (ASF) under one
```
Comment (hmrajas) [inline src/test/resources/artifacts/AXIS2/config/test_axis2_server_9000.xml]: Please use WSO2 licence header.

File: src/test/IntegrationTest.txt [hunk 7]
```diff
@@ -0,0 +1,41 @@
+Product: Integration tests for WSO2 ESB SMPP connector
+
+Pre-requisites:
+
+ - Maven 3.x
+ - Java 1.7
+ - The org.wso2.esb.integration.integration-base project is required. The test suite has been configured to download this project automatically.
+   If the automatic download fails, download the following project and compile it using the mvn clean install command to update your local repository:
+   https://github.com/wso2-dev/esb-connectors/tree/master/integration-base
```
Comment (hmrajas) [inline src/test/IntegrationTest.txt]: This should be https://github.com/wso2-extensions/esb-integration-base

**Pull request merged**